app-modules/notification/src/Notifications/Channels/MailChannel.php

```diff
@@ -0,0 +1,176 @@
+
+                    $deliverable->update([
+                        'delivery_status' => (! $isDemoModeEnabled)
+                            ? NotificationDeliveryStatus::Dispatched
+                            : NotificationDeliveryStatus::Successful,
+                        'quota_usage' => (! $isDemoModeEnabled)
+                            ? $this->determineQuotaUsage($result->recipients)
+                            : 0,
+                    ]);
+                } else {
+                    $deliverable->update([
+                        'delivery_status' => NotificationDeliveryStatus::DispatchFailed,
+                    ]);
+                }
+
+                // Consider dispatching this as a seperate job so that it can be encapsulated to be retried if it fails, but also avoid changing the status of the deliverable if it fails.
+                if ($notification instanceof HasAfterSendHook) {
+                    $notification->afterSend($notifiable, $deliverable, $result);
+                }
+            } catch (Throwable $exception) {
+                report($exception);
+            }
+        } catch (NotificationQuotaExceeded $exception) {
+            $deliverable->update(['delivery_status' => NotificationDeliveryStatus::RateLimited]);
+        } catch (Throwable $exception) {
+            $deliverable->update([
+                'delivery_status' => NotificationDeliveryStatus::DispatchFailed,
+            ]);
+
+            throw $exception;
+        }
+    }
+
+    protected function determineQuotaUsage(array $recipients): int
+    {
+        $users = User::query()
+            ->with('roles')
+            ->whereIn('email', array_map(
+                fn (Address $recipient): string => $recipient->getAddress(),
+                $recipients,
+            ))
+            ->get()
+            ->keyBy('email');
+
+        return collect($recipients)
+            ->filter(fn (Address $recipient): bool => ! ($users[$recipient->getAddress()] ?? null)?->isSuperAdmin())
+            ->count();
+    }
+
+    protected function canSendWithinQuotaLimits(MailMessage $message, OutboundDeliverable $deliverable): bool
+    {
+        $primaryRecipientUsage = ($deliverable->recipient instanceof User && $deliverable->recipient->isSuperAdmin()) ? 0 : 1;
+
+        // 1 for the primary recipient, plus the number of cc and bcc recipients
+        $estimatedQuotaUsage = $primaryRecipientUsage + count($message->cc) + count($message->bcc);
+
+        $licenseSettings = app(LicenseSettings::class);
+
+        $resetWindow = $licenseSettings->data->limits->getResetWindow();
+
+        $currentQuotaUsage = OutboundDeliverable::query()
+            ->where('channel', NotificationChannel::Email)
+            ->whereBetween('created_at', [$resetWindow['start'], $resetWindow['end']])
+            ->sum('quota_usage');
+
+        return ($currentQuotaUsage + $estimatedQuotaUsage) <= $licenseSettings->data->limits->emails;
+    }
```

Comment:
Where did we end up with the discussion of these being different? Did we want to split off fixing/unifying it into a different ticket?

Reply:
I don't think we decided, but technically they are doing things. First method is checking the admin status of all recipients (`to`, `cc`, `bcc`), second is checking the admin status of `to` and assuming `cc` and `bcc` are not admins. I think the second should probably always be used but this is your call.

Reply:
We talked about this in a meeting so we are all good here.